RecoJets/JetProducers/src/PileupJetIdAlgo.cc

```diff
@@ -566,52 +741,72 @@ PileupJetIdentifier PileupJetIdAlgo::computeIdVariables(const reco::Jet* jet,
   internalId_.nParticles_ = jet->nConstituents();
 
   ///////////////////////pull variable///////////////////////////////////
-  float sumW2(0.0);
-  float sum_deta(0.0), sum_dphi(0.0);
-  float ave_deta(0.0), ave_dphi(0.0);
-  for (size_t j = 0; j < jet->numberOfDaughters(); j++) {
-    const auto& part = jet->daughterPtr(j);
-    if (!(part.isAvailable() && part.isNonnull())) {
-      continue;
-    }
 
-    float partPuppiWeight = 1.0;
-    if (usePuppi) {
-      const pat::PackedCandidate* partpack = dynamic_cast<const pat::PackedCandidate*>(part.get());
-      if (partpack != nullptr)
-        partPuppiWeight = partpack->puppiWeight();
-    }
+  if (!useBugFix) {
+//    float sumW2(0.0);
+//    float sum_deta(0.0), sum_dphi(0.0);
+//    float ave_deta(0.0), ave_dphi(0.0);
+    for (size_t j = 0; j < jet->numberOfDaughters(); j++) {
+      const auto& part = jet->daughterPtr(j);
+      if (!(part.isAvailable() && part.isNonnull())) {
+        continue;
+      }
+
+      float partPuppiWeight = 1.0;
+      if (usePuppi) {
+        const pat::PackedCandidate* partpack = dynamic_cast<const pat::PackedCandidate*>(part.get());
+        if (partpack != nullptr) {
+          partPuppiWeight = partpack->puppiWeight();
+        }
+      }
 
-    float weight = (part->pt()) * partPuppiWeight;
-    float weight2 = weight * weight;
-    sumW2 += weight2;
-    float deta = part->eta() - jet->eta();
-    float dphi = reco::deltaPhi(*part, *jet);
-    sum_deta += deta * weight2;
-    sum_dphi += dphi * weight2;
-    if (sumW2 > 0) {
-      ave_deta = sum_deta / sumW2;
-      ave_dphi = sum_dphi / sumW2;
+      float weight = (part->pt()) * partPuppiWeight;
+      float weight2 = weight * weight;
+      sumW2 += weight2;
+      float deta = part->eta() - jet->eta();
+      float dphi = reco::deltaPhi(*part, *jet);
+      sum_deta += deta * weight2;
+      sum_dphi += dphi * weight2;
+      if (sumW2 > 0) {
+        ave_deta = sum_deta / sumW2;
+        ave_dphi = sum_dphi / sumW2;
+      }
```

Comment:
Do not recalculate the averages in every iteration inside the loop

Reply:
Will move it outside of the loop and fix also in the `master` branch.